### PR TITLE
Feature - track HashiCorp Vault dependency

### LINKED ISSUE #181: Provide capability to opt-in for dependency tracking for HashiCorp Vault
**Is your feature request related to a problem? Please describe.**
Provide capability to opt-in for dependency tracking for HashiCorp Vault

**Describe the solution you'd like**
```csharp
builder.AddHashiCorpVaultWithUserPass(
    // URI where the HashiCorp Vault is running.
    vaultServerUriWithPort: "https://uri.to.your.running.vault:5200",
    // Username/Password combination to authenticate with the vault.
    username: "admin",
    password: "s3cr3t",
    // Path where the secrets are stored in the KeyValue secret engine.
    secretPath: "my-secrets",

    useDependencyTracking: true
);
```

**Additional context**
We need to see if we can do this by bringing in that dependency or if that would cause circular references

### PR DESCRIPTION
**Pull request by stijnmoreels**
Adds dependency tracking to the HashiCorp Vault secret provider. Since this one is not yet released publicly, we can safely change the extensions and signatures of this one.
Made (like the Azure Key Vault) seperate options were optional values can be configured.

Closes #181

### CODE REVIEW

**Comment by stijnmoreels** [comment]
Same here? Serilog addition to track in Application Insights? 
Will wait for #207 so we can re-use it.

File: docs/preview/features/secret-store/provider/hashicorp-vault.md [hunk 1]
```diff
@@ -95,14 +95,21 @@ public class Program
                             secretPath: "my-secrets");
 
                         // Version of the KeyValue secret engine (default: V2).
-                         builder.AddHashiCorpVault(..., keyValueVersion: VaultKeyValueSecretEngineVersion.V1);
+                         builder.AddHashiCorpVault(..., options => options.KeyValueVersion: VaultKeyValueSecretEngineVersion.V1);
```
Comment (tomkerkhove) [inline docs/preview/features/secret-store/provider/hashicorp-vault.md:98]: Was this documentation outdated or breaking change?

**Review by tomkerkhove — APPROVED** [review]

File: docs/preview/features/secret-store/provider/hashicorp-vault.md (hunk 1 above)
Comment (tomkerkhove) [reply docs/preview/features/secret-store/provider/hashicorp-vault.md:98]: If this was not shipped yet, then it's ok

Comment (stijnmoreels) [reply docs/preview/features/secret-store/provider/hashicorp-vault.md:98]: Not only the documentation was changed, the extension and signature too. And yes, this isn't shipped yet in a minor version (ony previews) so I guess it's OK to still break this.

**Comment by tomkerkhove** [comment]
Feel free to merge when you are done @stijnmoreels!

**Pull request merged**